src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,140 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final String VALID_PASSWORD = "password";
+    private static final String INVALID_PASSWORD = "pswrd";
+    private static final String VALID_LOGIN = "Valid_login@gmail.com";
+    private static final String INVALID_LOGIN = "invalid_login";
+    private static final Integer VALID_AGE = 20;
+    private static final Integer INVALID_AGE = 12;
+    private static final Integer NEGATIVE_AGE = -10;
+    private static final Integer BIG_AGE = 150;
+    private static final Long ID = 1234L;
+    private static final Long NEGATIVE_ID = -5L;
+    private final RegistrationService registerService = new RegistrationServiceImpl();
+    private User testUser = new User();
+
+    @BeforeEach
+    void setUp() {
+        testUser.setPassword(VALID_PASSWORD);
+        testUser.setLogin(VALID_LOGIN);
+        testUser.setId(ID);
+        testUser.setAge(VALID_AGE);
+    }
+
+    @AfterEach
+    void tearDown() {
+        Storage.people.clear();
+    }
+
+    @Test
+    void userDataCorrect_ok() {
+        registerService.register(testUser);
+        boolean actual = Storage.people.contains(testUser);
+        assertTrue(actual);
+    }
+
+    @Test
+    void userAreNull_notOk() {
+        assertThrows(IncorrectDataExeption.class, () -> {
+            registerService.register(null);
+        }, "User can not be null");
+    }
+
+    @Test
+    void userAgeAreLessThat18_notOk() {
+        testUser.setAge(INVALID_AGE);
```

Comment:
for testing it is better to take marginally unacceptable input data, in this case 17 years